mlrun/model.py

```diff
@@ -756,6 +765,20 @@ def logs(self, watch=True, db=None):
             print(f"final state: {state}")
         return state
 
+    def wait_for_completion(self, sleep=3, timeout=0):
+        """wait for async run to complete"""
+        total_time = 0
+        while True:
+            state = self.state()
+            if state not in ["running", "pending"]:
+                break
+            time.sleep(sleep)
+            total_time += sleep
+            if total_time > timeout:
+                print("exit after timout")
+                break
```

Comment:
You should raise exception here, I think you can create a new timeout error for mlrun:
```
class MLRunTimeoutError(MLRunHTTPStatusError, TimeoutError):
    error_status_code = HTTPStatus.GATEWAY_TIMEOUT.value
```
and then here:
`raise mlrun.errors.MLRunTimeoutError(f"Run did not reach terminal state on time")`

Reply:
done